File: 2196. Create Binary Tree From Descriptions.py
class Solution:
    def createBinaryTree(self, descriptions: List[List[int]]) -> Optional[TreeNode]:
        store = defaultdict(TreeNode)
        parentStore, childStore = set(), set()

        for parent, child, direction in descriptions:
            parentStore.add(parent)
            childStore.add(child)

            if parent not in store:
                store[parent] = TreeNode(parent)
            
            if direction:
                if child not in store:
                    store[child] = TreeNode(child)
                store[parent].left = store[child]
            else:
                if child not in store:
                    store[child] = TreeNode(child)
                store[parent].right = store[child]
                
        for parent in parentStore:
            if parent not in childStore:
                return store[parent]

File: 2196. Create Binary Tree From Descriptions.cpp
class Solution {
public:
    map<int,vector<pair<int,int>>> store;
    void constructTree(TreeNode* node)
    {
        if(!node or store.find(node->val) == store.end()) return;

        for(auto child:store[node->val])
        {
            TreeNode* childNode = new TreeNode(child.first);
            if(child.second) node->left = childNode;
            else node->right = childNode;
            constructTree(childNode);
        }
    }
    TreeNode* createBinaryTree(vector<vector<int>>& des) 
    {
        map<int, int>count;
        for(auto dir:des)
        {
            int parent = dir[0], val = dir[1], isLeft = dir[2];
            if(not count[parent]) count[parent]++;
            count[val]++, count[val]++;
            store[parent].push_back({val, isLeft});
        }

        TreeNode* root = NULL;
        for(auto [val, key]:count)
        {
            if(key == 1)
            {
                root = new TreeNode(val);
                break;
            }
        }
        constructTree(root);
        return root;
    }
};